src/language/formatter/formatter.ts

```diff
@@ -55,6 +55,7 @@ export interface IFormatter {
      * Formats linter results
      * @param failures Linter failures that were not fixed
      * @param fixes Fixed linter failures. Available when the `--fix` argument is used on the command line
+     * @param fileNames All of file paths that were linted
      */
-    format(failures: RuleFailure[], fixes?: RuleFailure[]): string;
+    format(failures: RuleFailure[], fixes?: RuleFailure[], fileNames?: string[]): string;
```

Comment:
again, pretty sure the params should not be marked optional

Reply:
not sure what you mean or if this is still relevant. required arguments can't come after optional arguments. and we don't want to break this API.